lib/cli.rb

```diff
@@ -0,0 +1,164 @@
+        puts 
+
+        # writing out my idea 
+        # if same user and same search location show up more than 3 times (iterate over searches) 
+        # then spit out puts "Notice how actual user is looking at specific location"
+  
+          # if specific_searches > 5
+          #   puts "Looks like someone has been searching #{search.search_location} a lot.."      
+          # else
+          #   puts "No particular place seems to be of interest to anyone."
+          # end  
+      elsif choice == "3" 
+        puts`clear`
+        puts 
+        UsersSearch.all.each do |history|
+          puts "ID - #{history.id}, Name - #{history.user.name} and Location - #{history.search.search_location}."  
+          end 
+          puts 
+          puts "Are you sure you want to delete a specific search? You're being stormy..." 
+          puts 
+        puts "Please type in the specific ID you wish to delete"
+        puts 
+        delete_select = STDIN.gets.chomp.strip
+          if UsersSearch.find_by(id: delete_select)
+            UsersSearch.delete(delete_select)
+            puts 
+            puts "You're shady. Like the "+"weather. ".colorize(:cyan)+"Get it?"
+            puts 
+            puts "You have deleted ID #{delete_select}. I hope you're happy."
+            puts 
+            is_running = true
+          else 
+            puts 
+            puts "Please select 3 again. The ID you typed in doesn't exist."
+            puts
+            puts "Try using your eyes and reading the screen for once..."
+            puts
+          end 
+
+      elsif choice == "4" 
+        puts `clear`
+        Search.all.each do |search|
+          puts "Location: #{search.search_location}"  
+          end 
+        puts 
+        puts "Changing a search record's location. You are hiding something with clout.. get it.."
+        puts 
+        puts "Please type a location name you want to change"
+        puts  
+        user_picking_location = STDIN.gets.chomp.strip
+        location_on_table = Search.find_by(search_location: user_picking_location) 
+          if location_on_table 
+            puts "Please type in the NEW location name. Straight up changing history..."
+            user_new_location = STDIN.gets.chomp.strip
+            location_on_table.update(search_location: user_new_location)
+            puts 
+            puts "Your location changed from #{user_picking_location} to #{user_new_location}."
+            puts
+            puts "That's not sketchy at all..." 
+            puts 
+          else 
+            puts 
+            puts "Please select 4 again - the location you typed in doesn't exist"
+            puts 
+            is_running = true 
+        end 
+
+        # kicking me out of the program, want the program to stay up and running while Im still in it - 
+
+      elsif choice == "5"
+        puts `clear`
+        puts 
+        puts "Toodles"
+        puts 
+        is_running = false 
```

Comment:
```suggestion
        exit
```
Instead of setting a flag and then checking next iteration, do the work now.
[how-to-exit-a-ruby-program](https://www.honeybadger.io/blog/how-to-exit-a-ruby-program/)